sympy/solvers/tests/test_solveset.py

```diff
@@ -1363,13 +1349,14 @@ def test_trig_system():
     # TODO: add more simple testcases when solveset returns
     # simplified soln for Trig eq
     assert nonlinsolve([sin(x) - 1, cos(x) -1 ], x) == S.EmptySet
-    soln1 = (ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),)
-    soln = FiniteSet(soln1)
-    assert nonlinsolve([sin(x) - 1, cos(x)], x) == soln
+    # test for nonlinsolve([sin(x) - 1, cos(x)], x) in @xfail
 
 
 @XFAIL
 def test_trig_system_fail():
+    soln1 = (ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),)
+    soln = FiniteSet(soln1)
+    assert nonlinsolve([sin(x) - 1, cos(x)], x) == soln
```

Comment:
XFAIL tests should not be grouped together. The test runner runs these tests and reports when they pass. The idea is that if it passes we can remove the XFAIL to make it an actual test. If we group multiple tests in one XFAIL test then that doesn't work because the whole test won't pass until each individual part is separately fixed.

Reply:
Sure, I'll split them up.

BTW, how do I run flake8 locally? Somehow if I call `flake8 sympy` (from the top-level directory of my local git repository) I get lots and lots of errors/warnings (including for lines that have a `#noqa` comment, like those in sets/handlers/union). Due to this I can't figure out which messages are those that are actually relevant, i.e. those that will make Travis fail.

Reply:
The errors you see from flake8 depend on the Python version you are using. I think Travis uses Python 3.8 for the flake8 tests so you should too. In particular I think that depending on Python version the `# noqa` for multipledispatch either needs to be on the line of the decorator or the line of the function signature.

Reply:
> 
> The errors you see from flake8 depend on the Python version you are using. I think Travis uses Python 3.8 for the flake8 tests so you should too. In particular I think that depending on Python version the `# noqa` for multipledispatch either needs to be on the line of the decorator or the line of the function signature.

Thanks a lot! That should help me fix my dev setup.